mesonbuild/backend/ninjabackend.py

```diff
@@ -1784,8 +1784,16 @@ def generate_link(self, target, outfile, outname, obj_list, linker, extra_args=[
         commands += self.get_cross_stdlib_link_args(target, linker)
         commands += linker.get_linker_always_args()
         if not isinstance(target, build.StaticLibrary):
-            commands += compilers.get_base_link_args(self.environment.coredata.base_options,
-                                                     linker)
+            options = self.environment.coredata.base_options
+            old_b_lundef = None
+            if (isinstance(target, build.SharedLibrary) and
+                    target.allow_undefined is not None and
+                    'b_lundef' in options):
+                old_b_lundef = options['b_lundef'].value
+                options['b_lundef'].value = not target.allow_undefined
+            commands += compilers.get_base_link_args(options, linker)
+            if old_b_lundef is not None:
+                options['b_lundef'].value = old_b_lundef
```

Comment:
Possibly; they look pretty similar though, so I'm not sure there's any benefit to switching. I think the latter actually means allow undefined symbols in _referenced_ shared libraries, not the one that's being built (so you can link with a fake/empty library that's replaced at runtime.)